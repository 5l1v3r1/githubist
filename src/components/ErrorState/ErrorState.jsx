// @flow

import React from 'react';

import { ServerStatus } from '..';

import styles from './ErrorState.scss';

const ErrorState = () => (
  <ServerStatus code={500}>
    <div className={styles.error}>
      <p className={styles.text}>Bir hata meydana geldi.</p>
    </div>
  </ServerStatus>
);

export default ErrorState;
