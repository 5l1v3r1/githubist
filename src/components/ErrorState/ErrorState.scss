@import '../../theme/theme';

.error {
  padding: $section-gap / 2 0;
  text-align: center;
}

.text {
  color: $text-muted;
  font-size: map-get($font-sizes, large);
}
