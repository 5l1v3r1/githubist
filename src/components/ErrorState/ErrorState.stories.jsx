// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';

import ErrorState from './ErrorState';

storiesOf('ErrorState', module)
  .addDecorator(backgrounds([{ name: 'dark', value: '#2d2e33', default: true }]))
  .add('default', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <ErrorState />
    </div>
  ));
