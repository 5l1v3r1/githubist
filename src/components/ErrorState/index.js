// @flow

import ErrorState from './ErrorState';

export default ErrorState;
