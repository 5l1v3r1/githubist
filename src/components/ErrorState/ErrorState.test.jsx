// @flow

import React from 'react';
import { shallow } from 'enzyme';

import ErrorState from './ErrorState';

it('renders correctly', () => {
  const wrapper = shallow(<ErrorState />);

  expect(wrapper).toMatchSnapshot();
});
