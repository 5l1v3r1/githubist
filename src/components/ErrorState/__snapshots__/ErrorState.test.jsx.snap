// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Status
  code={500}
>
  <div>
    <p>
      Bir hata meydana geldi.
    </p>
  </div>
</Status>
`;
